packages/core/src/errors.ts

```diff
@@ -54,17 +56,23 @@ export class HttpError extends Error implements RetriableDecision {
   /** json error response */
   public json: any
 
+  public headers: HttpHeaders
+
   /* istanbul ignore next because of super() not being covered*/
   constructor(
     readonly statusCode: number,
     readonly statusMessage: string | undefined,
     readonly body?: string,
     retryAfter?: string | undefined | null,
     readonly contentType?: string | undefined | null,
-    message?: string
+    message?: string,
+    headers?: HttpHeaders
```

Comment:
Let's change type to: `HttpHeaders | undefined`:
```suggestion
    headers?: HttpHeaders | undefined
```